src/Parser/RestoDruid/Modules/Traits/Persistence.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import HealingDone from 'Parser/Core/Modules/HealingDone';
+import Rejuvenation from '../Features/Rejuvenation';
+
+
+/**
+ * Persistence (Artifact Trait)
+ * Increases the duration of Moonfire, Rejuvenation and Sunfire by 1s. We will only look at the Rejuvenation part.
+ *
+ * This modules checks whenever a rejuvenation is applied or refreshed and add that as 1s to the variable persistanceValue.
+ * We make use of what the average healing from one second of rejuvenation did.
+ * This doesn't take "magic" rejuvenations into account that doesn't trigger an apply/refreshbuff event.
+ * TODO: Improve this module:
+ * TODO: Add mastery healing, add cultivation (?),
+ * TODO: Subtractions for overheals on last tick, subtractions for early refreshes
+ * TODO: Include "magic rejuvenations" such as deep rooted.
+ */
+class Persistence extends Module {
+  static dependencies = {
+    combatants: Combatants,
+    healingDone: HealingDone,
+    rejuvenation: Rejuvenation,
+  };
+
+  rank = 0;
+  totalPersistanceValue = 0;
+
+  on_initialized() {
+    this.rank = this.combatants.selected.traitsBySpellId[SPELLS.PERSISTENCE_TRAIT.id];
+    this.active = this.rank > 0;
+  }
+
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if(SPELLS.REJUVENATION.id !== spellId && SPELLS.REJUVENATION_GERMINATION.id !== spellId) {
+      return;
+    }
+    this.totalPersistanceValue++;
+  }
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
+    if(SPELLS.REJUVENATION.id !== spellId && SPELLS.REJUVENATION_GERMINATION.id !== spellId) {
+      return;
+    }
+    this.totalPersistanceValue++;
+  }
+
+  getSpellUptime(spellId) {
+    return Object.keys(this.combatants.players)
+      .map(key => this.combatants.players[key])
+      .reduce((uptime, player) =>
+        uptime + player.getBuffUptime(spellId), 0);
+  }
+
+  subStatistic() {
+    const uptimeRejuvenation = Math.floor((this.getSpellUptime(SPELLS.REJUVENATION.id) + this.getSpellUptime(SPELLS.REJUVENATION_GERMINATION.id)) / 1000);
+    const totalRejuvenationHealing = this.rejuvenation.totalRejuvHealing;
+    const oneSecondRejuvenationHealing = totalRejuvenationHealing/uptimeRejuvenation;
+
+    const persistenceThroughput = this.owner.getPercentageOfTotalHealingDone(this.totalPersistanceValue * oneSecondRejuvenationHealing);
```

Comment:
Looks good. I'm working on enhancements to the Rejuvenation module to implement some of the optional things, but you should probably merge first.